### PR TITLE
Solve pylint issues in examples

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- Solve pylint issues in examples.
- Add a protocol for the sync api request method. It was needed to type one of the examples correctly.

### CODE REVIEW

File: examples/debug_info.py [hunk 1]
```diff
@@ -90,83 +95,83 @@
...
     print("Printing information about endpoints in the Gateway")
     data = api(gateway.get_endpoints())
     print(jsonify(data))
 
 
-def print_all_devices():
+def print_all_devices() -> None:
     """Print all devices as JSON."""
     print("Printing information about all devices paired to the Gateway")
     if len(devices) == 0:
```
Comment (ggravlingen) [inline examples/debug_info.py:130]:
```suggestion
    if not devices:
```

File: examples/debug_info.py [hunk 2]
```diff
@@ -90,83 +95,83 @@
...
+        container.append(dev.raw.dict())
     print(jsonify(container))
 
 
-def print_lamps():
+def print_lamps() -> None:
     """Print all lamp devices as JSON."""
     print("Printing information about all lamps paired to the Gateway")
     lights = [dev for dev in devices if dev.has_light_control]
     if len(lights) == 0:
```
Comment (ggravlingen) [inline examples/debug_info.py:143]:
```suggestion
    if not lights:
```

File: examples/debug_info.py [hunk 3]
```diff
@@ -90,83 +95,83 @@
...
+        container.append(light.raw.dict())
     print(jsonify(container))
 
 
-def print_smart_tasks():
+def print_smart_tasks() -> None:
     """Print smart tasks as JSON."""
     print("Printing information about smart tasks")
     tasks = api(gateway.get_smart_tasks())
     if len(tasks) == 0:
```
Comment (ggravlingen) [inline examples/debug_info.py:156]:
```suggestion
    if not tasks:
```

File: examples/debug_info.py (hunk 1 above)
Comment (MartinHjelmare) [inline examples/debug_info.py:130]:
```suggestion
    if not devices:
```

**Review by ggravlingen — CHANGES_REQUESTED** [review]
Great work here! Just a few minor changes suggested.

File: examples/debug_info.py [hunk 4]
```diff
@@ -90,83 +95,83 @@
...
+        container.append(dev.raw.dict())
     print(jsonify(container))
 
 
-def print_lamps():
+def print_lamps() -> None:
     """Print all lamp devices as JSON."""
     print("Printing information about all lamps paired to the Gateway")
     lights = [dev for dev in devices if dev.has_light_control]
     if len(lights) == 0:
```
Comment (MartinHjelmare) [inline examples/debug_info.py:143]:
```suggestion
    if not lights:
```

**Comment by MartinHjelmare** [comment]
Right. Yeah we can move it to the wiki.

**Comment by ggravlingen** [comment]
Where are they though? I seem to remember having a few pages with random code before I moved the library to this organisation.

**Comment by MartinHjelmare** [comment]
The wiki was disabled in the repo settings. I've enabled it now.

**Comment by ggravlingen** [comment]
I'll create the color example page so you can put your focus here.

**Comment by MartinHjelmare** [comment]
Thanks!

**Pull request merged**